src/controllers.lisp

```diff
@@ -116,15 +123,41 @@
     (let ((*draw-mode* nil))
       (call-next-method))))
 
-(defmethod kit.sdl2:textinput-event :after ((instance sketch) timestamp text)
-  (on-text instance text))
+(defmethod kit.sdl2:textinput-event :after ((instance sketch-window) timestamp text)
+  (on-text (sketch instance) text))
 
-(defmethod kit.sdl2:keyboard-event :after ((instance sketch) state timestamp repeat-p keysym)
+(defmethod kit.sdl2:keyboard-event :after ((instance sketch-window) state timestamp repeat-p keysym)
   (when (not repeat-p)
-    (on-key instance
+    (on-key (sketch instance)
             ;; Removing the ugly "SCANCODE-" prefix from the keyword
             ;; symbol that denotes the button.
             (intern (subseq (symbol-name (sdl2:scancode keysym))
                             +scancode-prefix-length+)
                     (find-package "KEYWORD"))
             state)))
+
+;;; Backward compatibility.
```

Comment:
FWIW this section seems to be in a wrong file -- IMHO it should be in `sketch.lisp` (with `kit.sdl2:idle-render` for example) -- it doesn't have much to do with controllers.

Other kit.sdl2's methods that probably should be forwarded: `other-event`; maybe gl-window's slot readers/accessors: `sdl-window`, `sdl-window-id`, `gl-context`, `render-enabled`.

Reply:
EDIT: `sdl-window-id` is a function, so of course doesn't need to be forwarded.

Reply:
I wasn't sure exactly what was meant by "controllers", I interpreted it as "input-handling". But yes, some of the events like `close-window` do not fall under that definition. Happy to move it to `sketch.lisp`!